contracts/docs/.gitignore

```diff
@@ -0,0 +1 @@
+book/
```

Comment:
what is in the generated book, and why do we want to ignore it?

Reply:
This gitignore is generated by the forge docs command. I had initially thought to check in the generated docs, but they include the git commit in the markdown files themselves, meaning that our approach of building the docs in CI and checking for git diffs would not work, since every commit to the repo would cause  a diff. Instead, I elected to add README instructions for generating the docs.